### ISSUE TITLE
IsUnique not added for unique indexes (fluent API)

### ISSUE THREAD

**Issue by qrjo** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#586]
I have some unique indexes in my database, and they are added in the entity's fluent API configuration. However, the `.IsUnique()` isn't added, so they aren't configured to be unique. This may be done on purpose, but I think there's a bug on [this line](https://github.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/blob/master/EntityFramework.Reverse.POCO.Generator/EF.Reverse.POCO.v3.ttinclude#L5382):
```C#
if (nullable && (indexesForName.All(x => x.IsPrimaryKey) || indexesForName.All(x => x.IsUnique) || indexesForName.All(x => x.IsUniqueConstraint)))
```

I think it should be:
```C#
if (!nullable && (indexesForName.All(x => x.IsUnique) || indexesForName.All(x => x.IsUniqueConstraint)))
```

Note the added `!` and I removed `indexesForName.All(x => x.IsPrimaryKey)` because it cannot be true due to earlier filtering.

I might be wrong, so please correct me if I am.

**Comment by sjh37** [comment]
Hi @qrjo Yes, it was not correct.
I've double checked and I think it should actually be:
```c#
if (indexesForName.All(x => x.IsUnique) || indexesForName.All(x => x.IsUniqueConstraint))
```

**Comment by sjh37** [comment]
This will be in the next release.

**Comment by sjh37** [comment]
Now released as part of v3.2.0
